src/librbd/api/Migration.cc

```diff
@@ -753,7 +753,7 @@ int Migration<I>::prepare() {
 
   BOOST_SCOPE_EXIT_TPL(&m_dst_image_ctx) {
     if (m_dst_image_ctx != nullptr) {
-      delete m_dst_image_ctx;
+      m_dst_image_ctx->state->close();
```

Comment:
@dillaman Actually, I don't see why m_dst_image_ctx should be closed here. It is opened only in `create_dst_image` and it makes sure to close it before exiting (and set m_dst_image_ctx to null). So it should be not null only if we fail earlier.

Reply:
Because a valid `dst_image_ctx` is always passed into the constructor for `Migration` -- it's not technically opened but the memory is allocated and therefore needs to be properly freed. The close state machine already handles the case where the image isn't actually opened yet when it's invoked.

Reply:
Though it seems like there is an old bug in `create_dst_image`, that is "prevent prepare from cleaning up the ImageCtx" should be done only after the image is successfully opened, not before.

Reply:
Ah. Indeed. Missed this close behavior. Thanks.